images/fcos-bastion-image/root/usr/share/bastion_services/ignitions/wipe_disks.bu

```
@@ -18,7 +18,7 @@ systemd:
         Wants=network-online.target
         [Service]
         Type=oneshot
-        ExecStartPre=-bash -c 'set -x; for lv in $(lvs --noheadings -o lv_name,vg_name 2>/dev/null | awk '{print $2"/"$1}'); do lvchange -an "/dev/$lv" || true; done; for vg in $(vgs --noheadings -o vg_name 2>/dev/null); do vgremove -f "$vg" || true; done; for pv in $(pvs --noheadings -o pv_name 2>/dev/null); do pvremove -ff -y "$pv" || true; done; for dev in $(lsblk -I8,259 -nd --output name); do wipefs -a /dev/$dev* || true; dd if=/dev/zero of="/dev/$dev" bs=1M count=100 status=none || true; done; set +x'
+        ExecStartPre=/bin/sh -c 'set -euo pipefail; /bin/echo -e "xx=============================xx\n  STARTING FULL DISK CLEANUP  \nxx=============================xx\n"; if /sbin/vgs >/dev/null 2>&1; then /bin/echo -e ">>> LVM VOLUME GROUPS FOUND. DEACTIVATNG AND REMOVING ALL LVM METADATA <<<"; /sbin/vgchange -a n || true; /sbin/vgremove -f -ff $(/sbin/vgs --noheadings -o vg_name) || true; /sbin/pvremove -ff -y $(/sbin/pvs --noheadings -o pv_name) || true; else /bin/echo -e ">>> NO LVM FOUND. PROCEEDING TO WIPEFS ON ALL DISKS <<<"; fi; for DEVICE in $(lsblk -o NAME,TYPE,MOUNTPOINT -d --json | jq -r ".blockdevices[] | select(.type==\"disk\" and .mountpoint==null and .name | test(\"vd[b-z]|sd[b-z]|nvme[1-9]n[1-9]$\")) | .name"); do DEVICE_PATH="/dev/${DEVICE}"; /bin/echo -e "--- Wiping signatures on disk: ${DEVICE_PATH} ---"; /sbin/wipefs -a -f "${DEVICE_PATH}" || true; done; /bin/echo -e "\nxx=============================xx\n  FULL DISK CLEANUP COMPLETE  \nxx=============================xx"'
```

The original script actually wipes absolutely everything, regardless of whether the disk is LVM or not.

- Attempts to remove all LVM metadata (LVs, VGs, PVs), but gracefully ignores errors
- Iterates through all block devices discovered by lsblk -I8,259
- Runs wipefs -a on every device
- Runs dd if=/dev/zero on every device to zero the first 100MB
- Does not filter disks
- Does not avoid OS/root disks

So even on non-LVM machines, the script fully destroys all partitions, filesystems, and bootloader blocks.

The new changes check whether any LVM volume groups exist and, if so, deactivate and remove all LVM metadata (LVs, VGs, and PVs). It uses lsblk --json with jq to identify only unmounted secondary disks such as sdb–sdz, vdb–vdz, and nvme1n1–nvme9n9. The OS/root disk and any mounted disks are intentionally excluded, which is not the expectation, we want everything to be wiped. It also avoids deeper destructive overwrites (no dd). The changes are good for adding clear logs for each step, showing which disks are targeted and when the cleanup is complete.